db/crud.go

```diff
@@ -1907,6 +1965,27 @@ func (db *DatabaseCollectionWithUser) resolveDocMerge(ctx context.Context, local
 	return mergedRevID, docHistory, nil
 }
 
+// resolveLocalWinsHLV will update remote doc's body and attachments to match the local doc, and return a new HLV for local wins
+func (db *DatabaseCollectionWithUser) resolveLocalWinsHLV(ctx context.Context, localDoc, remoteDoc *Document) (*HybridLogicalVector, error) {
+
+	docBodyBytes, err := localDoc.BodyBytes(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to retrieve local document body while resolving conflict: %w", err)
+	}
+
+	newHLV, err := localWinsConflictResolutionForHLV(ctx, localDoc.HLV, remoteDoc.HLV, localDoc.ID, db.dbCtx.EncodedSourceID)
+	if err != nil {
+		return nil, err
+	}
+
+	remoteDoc.RemoveBody()
+	remoteDoc.Deleted = localDoc.IsDeleted()
+	remoteDoc.SetAttachments(localDoc.Attachments().ShallowCopy())
```

Comment:
I think this is fine for now, but is it possible that resolving an HLV conflict will also mean that affects a revtree as well?

I suspect it will need to call https://github.com/couchbase/sync_gateway/blob/02429a56456ff283c6cdc7c6d4e155212e70eefc/db/crud.go#L1824-L1862 as well.